packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -777,11 +783,25 @@ class _AppBarState extends State<AppBar> {
       _scrollNotificationObserver!.removeListener(_handleScrollNotification);
       _scrollNotificationObserver = null;
     }
+    _notificationListenerStates.clear();
     super.dispose();
   }
 
   void _handleScrollNotification(ScrollNotification notification) {
     if (notification is ScrollUpdateNotification && widget.notificationPredicate(notification)) {
+      final RawGestureDetector? rawGestureDetector = notification.context?.widget as RawGestureDetector?;
```

Comment:
Why do we need the gesture detector?

Reply:
When we have two lists side by side, when scrolling through both and then returning one of the two to the top, the `scrolledUnderElevation` does not remain. However, since each list has its own `RawGestureDetector`, we can track each list individually to see if we can disable our `scrolledUnderElevation`.

Reply:
Hmm, it feels kind of brittle. ScrollPositions dispatch scroll notifications, it is possible that there is no RawGestureDetector if there is no Scrollable. Some folks have their own custom Scrollable classes. Is there another way we can differentiate multiple scrollables?

Reply:
I'm trying to look at this more clearly. From what I could tell, everything led me to RawGestureDetector as being what was behind it all. I tested several in one app and it worked very well.
But I agree that it feels kind of brittle.

Reply:
> RawGestureDetector as being what was behind it all

Do you mean the cause of the issue?

Reply:
I'm sorry, I expressed myself poorly. I'm talking more about interactions with the list and any scrollable widget.
There is always a RawGestureDetector that interacts with the widgets, so I decided to use it to observe the changes.

Reply:
Ok, no worries! Thank you for clarifying. :)

I don't think this is a reliable way to catch this. Would you like to consider a different approach?

Reply:
I made a small change, but it's still not ideal. I need to analyze and see if there really is a better way with what we have today or create something new.

Reply:
I made a change to observe the `ScrollableState` instead of the `Widget`(RawGestureDetector).